src/providers/solana/cachedRpcFactory.ts

```diff
@@ -62,6 +63,26 @@ export class CachedSolanaRpcFactory extends SolanaClusterRpcFactory {
     };
   }
 
+  getNetworkId(): Promise<number> {
+    return Promise.resolve(this.chainId);
+  }
+
+  async getBlock(blockNumber: string | number): Promise<unknown> {
+    // TODO: Support other evm block tags.
+    let slot;
+    try {
+      slot = BigInt(blockNumber);
+    } catch {
+      // Assume block tag = "latest";
+      slot = BigInt(await this.getBlockNumber());
+    }
+    return this.rateLimitedRpcClient.getBlock(slot, {});
+  }
```

Comment:
This could be simplified by narrowing the string type.

```suggestion
  async getBlock(blockNumber: number | "latest"): Promise<unknown> {
    const slot = blockNumbe === "latest"
      ? BigInt(await this.getBlockNumber())
      : BigInt(blockNumber);

    return this.rateLimitedRpcClient.getBlock(slot, {});
  }
```